app/models/manageiq/consumption/showback_event.rb

```diff
@@ -82,13 +82,12 @@ def get_key_flavor(key)
     data["FLAVOR"][data["FLAVOR"].keys.max][key]
   end
 
-  def update_event
-    #data_units = load_column_units
-    generate_data unless self.data.present?
+  def update_event(data_units = ManageIQ::Consumption::ConsumptionManager.load_column_units)
+    generate_data(data_units) unless self.data.present?
     @metrics = if  resource.methods.include?(:metrics) then metrics_time_range(end_time,start_time.end_of_month) else [] end
```

Comment:
@aljesusg Would we be able to generate events for past months?
@lpichler Is there any requirements you miss here?

Reply:
If we wanna generate events for past months We need to do it in another PR @sergio-ocon, It's not difficult do it, we only need to get all metrics of the month from CU and make the data object.

Reply:
Let's see if it is needed, @lpichler can tell us

Reply:
We can always add a PR if we need it

Reply:
+1 for `past months`, so in other PR, I am late for the party